src/components/App.tsx

```diff
@@ -1054,6 +1062,37 @@ class App extends React.Component<AppProps, AppState> {
     });
   }, SCROLL_TIMEOUT);
 
+  private autosaveLocalSceneDebounced = debounce(
+    async (elements: readonly ExcalidrawElement[], state: AppState) => {
+      if (this.state.autosave && this.state.fileHandle && supported) {
+        try {
+          await saveAsJSON(
+            elements,
+            state,
+            // only if fileHandle valid
+            true,
+          );
+        } catch (error) {
+          this.setState({
+            autosave: false,
+            toastMessage:
+              error.name === "NotAllowedError"
+                ? t("toast.autosaveFailed_notAllowed")
+                : error.name === "NotFoundError"
+                ? t("toast.autosaveFailed_notFound")
+                : t("toast.autosaveFailed"),
```

Comment:
suggestion: `if else` might be more readable here instead of nested ternary operators.

Reply:
Would require 3 if/else branches and duplicating the `setState` data which will be error-prone.

Reply:
I meant something like 👇 
```js
const toastMessage = t("toast.autosaveFailed");
if (error.name === "error.name === "NotAllowedError"") {
  toastMessage = t("toast.autosaveFailed_notAllowed");
}
else if (error.name === "NotFoundError") { 
  toastMessage = t("toast.autosaveFailed");
}
this.setState({ autosave: false, toastMessage});
```